test/spmd/tensor/test_elementwise_ops.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+from torch.testing._internal.common_utils import run_tests
+from ..test_utils import DistTensorTestBase, with_comms, TEST_GPU_NUM
+from spmd import distribute_tensor, DeviceMesh, Tensor, Shard, Replicate, _Partial
+from torch.distributed.distributed_c10d import ReduceOp
+from torch.testing._internal.common_distributed import (
+    skip_if_lt_x_gpu,
+)
+
+
+class DistElementwiseOpsTest(DistTensorTestBase):
+    # TODO: We need to add CPU tests for ops in the future.
+    def _run_sharded_elementwise_ops_gpu(
+        self, mesh, spec, input_size, op, reset_seed=None, **kwargs
+    ):
+        torch.manual_seed(self.rank)
+        input_tensor = torch.randn(*input_size, requires_grad=True).cuda(self.rank)
+        dist_tensor = Tensor.from_local(input_tensor, mesh, spec)
+        reset_seed() if reset_seed else None
+        dt = op(dist_tensor, **kwargs)
+        reset_seed() if reset_seed else None
+        expected = op(input_tensor, **kwargs)
+        self.assertEqual(input_tensor, dist_tensor.local_tensor())
+        self.assertEqual(expected, dt.local_tensor())
+
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_activations(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        self._run_sharded_elementwise_ops_gpu(device_mesh, [Shard(0)], (8, 5), torch.nn.functional.gelu)
```

Comment:
did we figure out why gelu/relu works magically

Reply:
No idea tbh..